discovery_engine_core/core/src/engine.rs

```diff
@@ -54,6 +56,34 @@ pub enum Error {
     Errors(Vec<Error>),
 }
 
+pub struct Config<'c, P> {
+    smbert: SMBertConfig<'c, P>,
+    kpe: KpeConfig<'c>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct StateData {
+    engine_state: Vec<u8>,
+    ranker_state: Vec<u8>,
+}
+
+pub fn create_engine<'c, P>(
```

Comment:
@finiteprods this also intersect a bit your story.

Reply:
I removed it from the current pr. I will tackle it in [TY-2309](https://xainag.atlassian.net/browse/TY-2309)
@finiteprods feel free to merge your pr #95 first

Reply:
ok sure. I will mark #95 ready and tag you guys.